skyportal/models/analysis.py

```diff
@@ -152,3 +187,196 @@ def authinfo(self):
     @authinfo.setter
     def authinfo(self, value):
         self._authinfo = value
+
+
+class AnalysisMixin:
+    def calc_hash(self):
```

Comment:
Many of these methods are borrowed from Guy's photseries work. Will be more fleshed out/tested in PR 3.

Reply:
Probably want it at the analysis level -- I can imagine wanting allow only certain groups to see some plots for some analysis and not for others.